codalab/lib/bundle_cli.py

```diff
@@ -717,14 +717,19 @@ def parse_client_worksheet_uuid(self, spec):
             client, worksheet_uuid = self.manager.get_current_worksheet_uuid()
         else:
             client_is_explicit = spec_util.client_is_explicit(spec)
-            client, spec = self.parse_spec(spec)
+            client, parsed_spec = self.parse_spec(spec)
             # If we're on the same client, then resolve spec with respect to
             # the current worksheet.
             if client_is_explicit:
                 base_worksheet_uuid = None
             else:
                 _, base_worksheet_uuid = self.manager.get_current_worksheet_uuid()
-            worksheet_uuid = self.resolve_worksheet_uuid(client, base_worksheet_uuid, spec)
+            try:
+                worksheet_uuid = self.resolve_worksheet_uuid(
+                    client, base_worksheet_uuid, parsed_spec
+                )
+            except ValueError:
+                raise UsageError('Invalid spec: "{}"'.format(spec))
```

Comment:
Invalid spec differs from a non-existent spec, right? Does this capture both? Is there any way to separate out those 2 failure cases?

Reply:
Hmm, could you give an example of invalid spec other than non-existent spec? IMO, `invalid spec == non-existent spec`. as there is no other invalid spec, but I could be wrong. On the other hand, if you feel it better be worded as "invalid spec", just let me know and we can change that.

Reply:
I guess I think of invalid = the spec string is malformed or otherwise non-comforming, whereas a non-existent spec is, say, doing `cl work somews::`, where `somews` is not a defined alias.

So right now, if you do:

```
$ cl work someworksheetthatdoesntexist18201
NotFoundError: No worksheet found with name 'someworksheetthatdoesntexist18201'
```

This diff shouldn't change this behavior, right?

Reply:
I see. Do we have definition of what a valid spec is, for example, must be alphanumerics? In that case, we can do a regex match to distinguish NotFoundError and InvalidSpecError. Otherwise, we would need to have them categorized as one error as there is no way to differentiate them.

Reply:
ah oops, i thought you might have an idea of that---no worries, then, i think things are fine as-is

Reply:
Nah I don't :) But we can always make future improvements.